src/bootstrap/src/core/build_steps/perf.rs

```diff
@@ -1,12 +1,115 @@
+use std::fmt::{Display, Formatter};
 use std::process::Command;
 
 use crate::core::build_steps::compile::{Std, Sysroot};
 use crate::core::build_steps::tool::RustcPerf;
 use crate::core::builder::Builder;
 use crate::core::config::DebuginfoLevel;
 
+/// Performs profiling or benchmarking with [`rustc-perf`](https://github.com/rust-lang/rustc-perf)
+/// using a locally built compiler.
+#[derive(Debug, Clone, clap::Parser)]
+pub struct PerfArgs {
+    #[clap(subcommand)]
+    cmd: PerfCommand,
+
+    #[clap(flatten)]
+    opts: SharedOpts,
+}
+
+impl Default for PerfArgs {
+    fn default() -> Self {
+        Self { cmd: PerfCommand::Eprintln, opts: SharedOpts::default() }
```

Comment:
Shouldn't we run all the benchmarks by default with something like `PerfCommand::All` when the user doesn't specify a particular benchmark module (e.g., `x perf`)?

Reply:
Just to clarify, these are individual profilers (`eprintln`, `cachegrind`, etc.), not benchmarks. I don't think that it makes sense to run all of them at once, it's like running `cargo check/build/test` at once, they just do very different things and serve different purposes. We could select some default, but I'm not sure if there even *is* a default profiler.

Once we actually add benchmarking in the future, I would propose `x perf` to just run benchmarks, but since we only have profilers now, I would opt into requiring users to select the profiler that they want to use.

Reply:
IMO it makes more sense to run `samply` (perhaps even together with `cachegrind`) rather than running `eprintln` by default. Currently the `eprintln` default doesn't really provide a good overall report about the compiler.

Reply:
So, just to clarify, I would prefer to keep the current situation, where you cannot just do `x perf`, but you have to select some profiler. Because currently I don't think there is a good default - for both `samply` and `cachegrind`, you need to have an external tool configured, so that isn't IMO a good default.

This `Default` implementation should be basically `unreachable!`.